code/parse/parsehi.cpp

```diff
@@ -17,3 +19,42 @@ bool parse_optional_float_into(const SCP_string& field_name, float* value_target
 	}
 	return false;
 }
+
+/**
+ * @brief Parses an optional table value into a field if the name is found
+ *
+ * @param field_name The name of the table field
+ * @param value_target Pointer to the variable to assign any parsed value to
+ *
+ * @return True if a value was parsed, false if not
+ */
+bool parse_optional_bool_into(const SCP_string& field_name, bool* value_target)
+{
+	if (optional_string(field_name.c_str())) {
+		stuff_boolean(value_target);
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Parses an optional table color into an object if the color is found
+ *
+ * @param field_name The name of the table field
+ * @param value_target Pointer to the variable to assign any parsed value to
+ *
+ * @return True if a value was parsed, false if not
+ */
+bool parse_optional_color3i_into(const SCP_string& field_name, hdr_color* out_color)
+{
+	if (optional_string(field_name.c_str())) {
+		int components[3] = {255, 255, 255};
+		stuff_int_list(components, 3, RAW_INTEGER_TYPE);
+		if (out_color == nullptr) {
+			out_color = new hdr_color();
+		}
```

Comment:
This isn't going to do anything for `out_color` because the calling function won't have the address of the allocated memory.  To make this work you would need a pointer-to-pointer.  However, given the risky memory management here, I would recommend taking this out and just adding an Assertion that the pointer isn't null.